### PR TITLE
Upgrade from Chromium 123 to Chromium 124.

### PR DESCRIPTION
**Pull request by mkarolin**
Resolves https://github.com/brave/brave-browser/issues/36346

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by boocmp — APPROVED** [review]
Speedreader & Tor LGTM

**Review by tmancey — APPROVED** [review]
LGTM++

**Review by zenparsing — APPROVED** [review]
Rewards 👍

File: android/java/res/xml/manage_sync_preferences.xml
Comment (AlexeyBarabash) [inline android/java/res/xml/manage_sync_preferences.xml:65]: :+1:

**Review by AlexeyBarabash — APPROVED** [review]
Sync ++.

Requires updated server https://github.com/brave/go-sync/pull/221 .

Tested basic sync work desktop <=> Android with updated local sync server - works fine.

File: browser/tor/brave_tor_browsertest.cc
Comment (darkdh) [inline browser/tor/brave_tor_browsertest.cc:154]: leftover comment? Also in `browser/ui/webui/brave_wallet/wallet_panel_ui_browsertest.cc`, `browser/ui/webui/settings/brave_extensions_manifest_v2_browsertest.cc` and `browser/ui/webui/settings/brave_tor_snowflake_extension_browsertest.cc`

**Review by darkdh — APPROVED** [review]
++ with one nit

File: browser/tor/brave_tor_browsertest.cc
Comment (mkarolin) [reply browser/tor/brave_tor_browsertest.cc]: Thanks @darkdh, fixed in https://github.com/brave/brave-core/pull/22320/commits/392096b2e4736035853b803866a581d95c2be4f5

File: ios/browser/api/qr_code/qr_code_generator.mm
Comment (kylehickinson) [inline ios/browser/api/qr_code/qr_code_generator.mm:88]: Is this the new default? Anything we need to do here?

**Review by kylehickinson — APPROVED** [review]

File: ios/browser/api/qr_code/qr_code_generator.mm
Comment (mkarolin) [reply ios/browser/api/qr_code/qr_code_generator.mm:88]: This option keeps it consistent with the prior behavior where the caller was responsible for creating "quiet zone" margins around the QR bitmap. The other, new option is `kIncluded` which would cause GenerateBitmap to add the margins to the QR bitmap.

It seems like upstream has a preference for kIncluded eventually:
```
+  // TODO(https://crbug.com/325664342): Audit callers of `GenerateBitmap` and
+  // see if they can/should use `kIncluded` instead (testing if the callers are
+  // okay with differently-sized images and margins).  Note that the quiet zone
+  // may help to detect the QR codes even for small codes.   Once all the users
+  // of `kWillBeAddedByClient` are removed, the `QuietZone` enum can be removed
+  // altogether.
```
So we may want to try to use kIncluded and remove our own margins to see if that works well.

File: browser/ui/webui/brave_wallet/wallet_panel_ui.cc
Comment (supermassive) [inline browser/ui/webui/brave_wallet/wallet_panel_ui.cc:97]: Why this is no longer needed?

Comment (cdesouza-chromium) [reply browser/ui/webui/brave_wallet/wallet_panel_ui.cc:97]: Upstream has deleted this function [1] as not needed anymore. I had tried to reinsert it through a clever use of overrides, however I wasn't sure this was at all required, as none of the tests involved with this call were broken. This change is still marked as WIP because I couldn't get to a final decision on this.

[1] https://crrev.com/c/5328987

Comment (supermassive) [reply browser/ui/webui/brave_wallet/wallet_panel_ui.cc:97]: > DisableContentSecurityPolicy was unused outside of tests (where it wasn't needed)

Not sure how to track implementation of "where it wasn't needed" part, but that seems fine if tests are green

File: test/BUILD.gn
Comment (bridiver) [inline test/BUILD.gn:1131]: we are not supposed to be adding new tests to this list. Why can't this be implemented as a `PlatformBrowserTest`?

File: browser/resources/settings/brave_wallet_page/add_wallet_network_dialog.html
Comment (bridiver) [inline browser/resources/settings/brave_wallet_page/add_wallet_network_dialog.html:40]: why are we using a chromium os file here?

File: patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch
Comment (bridiver) [inline patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch:9]: we should not be changing all private methods to protected. What specifically do we here?

File: chromium_src/third_party/blink/renderer/bindings/modules/v8/v8_storage_estimate.h
Comment (bridiver) [inline chromium_src/third_party/blink/renderer/bindings/modules/v8/v8_storage_estimate.h:23]: does anyone know what the original purpose of this was? It would be helpful if that was captured here

File: third_party/rust/chromium_crates_io/Cargo.toml
Comment (bridiver) [inline third_party/rust/chromium_crates_io/Cargo.toml]: if you're not setting an exact version I don't think you need this at all

File: chromium_src/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.h
Comment (bridiver) [inline chromium_src/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.h:18]: have we filed this as a bug with upstream?

File: chromium_src/third_party/blink/renderer/core/frame/local_frame.cc
Comment (bridiver) [inline chromium_src/third_party/blink/renderer/core/frame/local_frame.cc:117]: I don't believe this functionality is even relevant anymore cc @diracdeltas

Comment (diracdeltas) [reply chromium_src/third_party/blink/renderer/core/frame/local_frame.cc:117]: discussed on slack. we do still allow users to enable specific scripts when they block scripts on a page. but it's hard to find so i'm not against just removing this functionality.

File: browser/ui/views/page_action/brave_page_action_icon_container_view.cc
Comment (bridiver) [inline browser/ui/views/page_action/brave_page_action_icon_container_view.cc:66]: we should drop this buildflag since it defaults to true for all platforms

File: patches/chrome-browser-ui-views-tabs-tab.cc.patch
Comment (bridiver) [inline patches/chrome-browser-ui-views-tabs-tab.cc.patch:8]: `#define set_y(POINT) set_y(contents_rect.y() + Center(contents_rect.height(), gfx::kFaviconSize))`

Comment (bridiver) [reply patches/chrome-browser-ui-views-tabs-tab.cc.patch:8]: actually that is maybe a bit risky so I guess this patch is ok

File: patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch
Comment (bridiver) [reply patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch:9]: actually why aren't we just making these changes in `WebUIBubbleDialogView`? We can do a chromium_src override for `SetLayoutManager` and do the customization there. The current approach is extremely difficult to follow and seems unnecessarily complicated

Comment (bridiver) [reply patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch:9]: looking at this again I'm confused because we apparently have two different styles of webui bubbles? Why would we do that? Don't we want consistent styling of either always rounded corners or never? Why do only a few different things use BraveWebUIBubbleManager?

Comment (bridiver) [reply patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch:9]: cc @zenparsing

Comment (bridiver) [reply patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch:9]: also it's still not clear to me what is private here that needs to be protected

File: test/BUILD.gn
Comment (bridiver) [reply test/BUILD.gn:1131]: @AlexeyBarabash I thought we got rid of this and were only using the exceptions list now which has code owner?

Comment (bridiver) [reply test/BUILD.gn:1131]: I guess tests that have functionality guarded by `!IS_ANDROID` would still go here, but I don't think that is the case for most of these

File: browser/resources/settings/brave_wallet_page/add_wallet_network_dialog.html
Comment (bridiver) [reply browser/resources/settings/brave_wallet_page/add_wallet_network_dialog.html:40]: I think we just want to drop `cros` as it makes no sense here

Comment (cdesouza-chromium) [reply browser/resources/settings/brave_wallet_page/add_wallet_network_dialog.html]: Fixed in 2c31c21eb86c5ac254fbf15f6bd375e7ea7c2423.

File: patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch
Comment (cdesouza-chromium) [reply patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch:9]: To clarify about the `protected` patch, this patch was only introduced because we have a few places where we call `CreateWebUIBubbleDialog` inside the derived class, which is something upstream doesn't seem to want anymore. I was going to have @simonhong reviewing this, but at some point this fell through the things that needed to be checked.

Example error.

```
../../brave/browser/ui/views/brave_shields/cookie_list_opt_in_bubble_host.cc:103:52: error: 'CreateWebUIBubbleDialog' is a private member of 'WebUIBubbleManagerImpl<CookieListOptInUI>'
  103 |         WebUIBubbleManagerImpl<CookieListOptInUI>::CreateWebUIBubbleDialog(
      |                                                    ^
../../../src/chrome/browser/ui/views/bubble/webui_bubble_manager.h:152:40: note: declared private here
  152 |   base::WeakPtr<WebUIBubbleDialogView> CreateWebUIBubbleDialog(
      |              
```

File: chromium_src/chrome/browser/ui/webui/settings/site_settings_helper.cc
Comment (goodov) [inline chromium_src/chrome/browser/ui/webui/settings/site_settings_helper.cc:104]: pls add trailing comma.

File: browser/ui/views/page_action/brave_page_action_icon_container_view.cc
Comment (mkarolin) [reply browser/ui/views/page_action/brave_page_action_icon_container_view.cc:66]: Added a follow up issue https://github.com/brave/brave-browser/issues/37364

File: chromium_src/third_party/blink/renderer/bindings/modules/v8/v8_storage_estimate.h
Comment (goodov) [reply chromium_src/third_party/blink/renderer/bindings/modules/v8/v8_storage_estimate.h:23]: https://github.com/brave/brave-browser/issues/12370

File: patches/components-content_settings-core-common-content_settings_types.mojom.patch
Comment (goodov) [inline patches/components-content_settings-core-common-content_settings_types.mojom.patch]: I think this enum now can be extended via `chromium_src` override.
`chromium_src/components/content_settings/core/common/content_settings_types.mojom`
```
[BraveExtend]
enum ContentSettingsType {
  BRAVE_START,
  BRAVE_ADS = BRAVE_START,
  BRAVE_COSMETIC_FILTERING,
  BRAVE_TRACKERS,
  BRAVE_HTTP_UPGRADABLE_RESOURCES,
  BRAVE_FINGERPRINTING_V2,
  BRAVE_SHIELDS,
  BRAVE_REFERRERS,
  BRAVE_COOKIES,
  BRAVE_SPEEDREADER,
  BRAVE_ETHEREUM,
  BRAVE_SOLANA,
  BRAVE_GOOGLE_SIGN_IN,
  BRAVE_HTTPS_UPGRADE,
  BRAVE_REMEMBER_1P_STORAGE,
  BRAVE_LOCALHOST_ACCESS,
};
```

File: components/playlist/browser/playlist_background_web_contentses.cc
Comment (goodov) [inline components/playlist/browser/playlist_background_web_contentses.cc:35]: `base::NoDestructor`

**Review by goodov — APPROVED** [review]

File: chromium_src/chrome/browser/ui/webui/settings/site_settings_helper.cc
Comment (cdesouza-chromium) [reply chromium_src/chrome/browser/ui/webui/settings/site_settings_helper.cc]: Fixed in f35be2835bd80a36b72cc1b210dfcffae55f1c83.

File: chromium_src/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.h
Comment (mkarolin) [reply chromium_src/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.h:18]: I am not sure how to reproduce it in Chrome to file an issue with upstream, tbh.

File: components/playlist/browser/playlist_background_web_contentses.cc
Comment (mkarolin) [reply components/playlist/browser/playlist_background_web_contentses.cc]: Fixed in [6a9e985](https://github.com/brave/brave-core/pull/22320/commits/6a9e98567947495cd9127da25a3033c08054c009)

File: chromium_src/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.h
Comment (bridiver) [reply chromium_src/chrome/browser/tpcd/heuristics/redirect_heuristic_tab_helper.h:18]: isn't it enabled by a buildflag or feature flag? Disabling it in chrome should have the same effect?

File: patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch
Comment (bridiver) [reply patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch:9]: in either case we should definitely do a follow-up to look at this in general because it seems very odd to me that we have two different styles of bubbles

File: chromium_src/third_party/blink/renderer/bindings/modules/v8/v8_storage_estimate.h
Comment (bridiver) [reply chromium_src/third_party/blink/renderer/bindings/modules/v8/v8_storage_estimate.h:23]: can we add a link/comment here since this is going to be hard to track with git blame?

Comment (mkarolin) [reply chromium_src/third_party/blink/renderer/bindings/modules/v8/v8_storage_estimate.h:25]: Added a comment in 5f260d9a9ba

File: chromium_src/third_party/blink/renderer/core/frame/local_frame.cc
Comment (bridiver) [reply chromium_src/third_party/blink/renderer/core/frame/local_frame.cc:117]: sounds like we are going to check usage with p3a before disabling so we can leave this for now

File: test/BUILD.gn
Comment (mkarolin) [reply test/BUILD.gn]: Updated in 1ae6c846810 + fixup c51a389d8ec

File: patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch
Comment (mkarolin) [reply patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch:9]: Created a follow up issue https://github.com/brave/brave-browser/issues/37366

Comment (cdesouza-chromium) [reply patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch]: Pushed 4769dccc28a6954c56795efc6327e071fd922531 which sets `CreateWebUIBubbleDialog` as the only protected method.

**Review by bsclifton — APPROVED** [review]
jsdeps + string reviewers ++

File: browser/ui/views/frame/vertical_tab_strip_region_view.cc
Comment (iefremov) [inline browser/ui/views/frame/vertical_tab_strip_region_view.cc:116]: this is no longer true? The whole thing seems strange (i.e. having CHECK under if DCHECK), but still wondering what's happened

Comment (mkarolin) [reply browser/ui/views/frame/vertical_tab_strip_region_view.cc:116]: From what I recall this was now being called with 0 width/height. I am assuming some layouting change that wasn't happening before. @sangwoo108 thought it was ok to remove, I believe.

File: browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc
Comment (iefremov) [inline browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc:85]: just to make sure that @diracdeltas is ok with any csp changes

**Review by iefremov — APPROVED** [review]

File: browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc
Comment (diracdeltas) [reply browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc:85]: this was copied from https://github.com/brave/brave-core/blob/4a7de952e5c0087aa6a3a9b702e794b770ebdbed/browser/ui/webui/brave_wallet/wallet_panel_ui.cc#L80-L86 ? why was this part of the chromium update?

Comment (diracdeltas) [reply browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc:85]: i think we should remove https://assets.cgproxy.brave.com from this unless it's confirmed to be necessary

Comment (samartnik) [reply browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc]: Yes, it was excessive, pushed the fix https://github.com/brave/brave-core/commit/72026c23aa87a718720eb039c746053545f4ee6b

File: browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc
Comment (diracdeltas) [inline browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc:85]: why is this last line needed?

also it seems the default is `chrome://favicon2` now instead of `chrome://favicon` but probably we don't need either of them

Comment (mkarolin) [reply browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc:85]: cc: @samartnik

Comment (samartnik) [reply browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc]: The last line is the second param (`separator`) for `base::JoinString`. `chrome://favicon` and `chrome://image` are not required, removed them here https://github.com/brave/brave-core/commit/13ebf4e7b805514e624d5e3d3c16a4c7c66d4a41

**Review by bridiver — APPROVED** [review]

File: browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc
Comment (diracdeltas) [reply browser/ui/webui/brave_wallet/android/android_wallet_page_ui.cc]: ah sorry i missed that it wasn't part of the first parameter. lgtm.

**Comment by mkarolin** [comment]
**When uplifting also uplift these:**
* https://github.com/brave/brave-core/pull/22986

File: patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch
Comment (zenparsing) [reply patches/chrome-browser-ui-views-bubble-webui_bubble_manager.h.patch]: It looks like `WebUIBubbleManager` has been significantly changed recently, which is not surprising since the API was not ideal. Hopefully our customizations can be simplified now.

**Pull request merged**
